index.js

```diff
@@ -840,12 +840,20 @@ Auth0.prototype.loginPhonegap = function (options, callback) {
       return ref.close();
     }
 
-    if (result.id_token) {
-      setTimeout(function() { callback(null, _this._prepareResult(result)) }, 0);
+    // parseHash returns idToken, NOT id_token
+    // if (result.id_token) {
+    //   setTimeout(function() { callback(null, _this._prepareResult(result)) }, 0);
+    //   answered = true;
+    //   return ref.close();
+    // }
+
+    if (result.idToken) {
+      callback(null, result);
```

Comment:
The problem was not in the setTimeout, it was the result.id_token (which was wrong) and the prepareResult was not needed because the tokenPayload was already decoded in the parse function above